docs/Dockerfile

```diff
@@ -5,7 +5,7 @@ MAINTAINER Mu Li <muli@cs.cmu.edu>
 # First, build MXNet binaries (ref mxnet/docker/cpu/Dockerfile)
 #
 
-RUN apt-get update && apt-get install -y build-essential git libopenblas-dev libopencv-dev
+RUN apt-get update && apt-get install -y build-essential git libopenblas-dev liblapack-dev libopencv-dev
```

Comment:
Not sure whether I fully understand all of this. Let me try to summarize.

Amalgamation build: This build is not-affected with the current set of changes. It has it's own (unchanged) Makefile and that doesn't set the flag to include LAPACK. There is no method currently to change this behavior by a config, which is IMO the right thing to do as Amalgamation by today only supports building one specific inference model and not all types of potential applications. 

Ubuntu 14.04 does not have lapack mangled into openblas. It's separate. But it is part of the official Ubuntu packages for 14.04: (https://packages.ubuntu.com/search?keywords=liblapack-dev). And whether this package is the original netlib version or something mangled into openBlas-library doesn't really matter for us as our internal wrapper code can work with all of them. So it's not clear to me why we shouldn't use it for 14.04 as well

It may be necessary to put in a switch that for Ubuntu 14.04 we add a "-llapack" and for 16.04 we don't. But that should be straightforward to achieve. 

Generally, we can have a global "USE_LAPACK" in the config file. I think it makes a lot of sense. But IMO, this should be on by default whenever we also use blas/atlas/apple/openblas. I expect that we will have an increasing number of functionalities based on lapack, once we add the basic support. And then it will be hard to track and document what functionality needs lapack and what not. In doubt this will be visible only after a user tries something and then gets a runtime error which is annoying.